src/pages/orderpage/ReceiverInfoSection.tsx

```diff
@@ -0,0 +1,174 @@
+/** @jsxImportSource @emotion/react */
+import { useFormContext, useFieldArray } from "react-hook-form";
+import styled from "@emotion/styled";
+import ReceiverModal from "@/pages/orderpage/ReceiverModal";
+import { useState } from "react";
+import type { FullOrderFormValues } from "@/utils/validator";
+
+const ReceiverInfoSection = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [initialReceivers, setInitialReceivers] = useState<
+    { name: string; phone: string; quantity: number }[]
+  >([]);
+  const { control } = useFormContext<FullOrderFormValues>();
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "receivers",
+  });
+
+  const handleClickAdd = () => {
+    setInitialReceivers([...fields]);
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleSubmitReceiverData = (data: {
+    receivers: { name: string; phone: string; quantity: number }[];
+  }) => {
+    for (let i = fields.length - 1; i >= 0; i--) {
+      remove(i);
+    }
+    data.receivers.forEach((receiver) => append(receiver));
```

Comment:
RecieverModal과 동적으로 폼 input을 추가하기 위해 `useFieldArray`를 사용하셨죠.
그런데 상위 폼에서도 제출받은 receivers를 `useFieldArray`로 관리하니 코드 복잡도가 조금 높아지는 것 같네요.
`ReceiverInfoSection`에서는 동적으로 폼 input을 추가할 필요가 없어서 하나의 필드에 setValue로 반영해주기만 해도 충분할 것 같아요.

```suggestion
    setValue("receivers", data.recievers, )
```

객체 배열을 필드 value로 했을 때 성능상 문제가 없는지 체크하며 리팩터링해주세요!